shipcat_cli/src/status.rs

```diff
@@ -0,0 +1,362 @@
+use crate::{Result, Error, ErrorKind, Manifest};
+use serde::Serialize;
+use serde_json::json;
+use chrono::{Utc};
+
+use kube::{
+    api::{Api, Object, PatchParams},
+    client::APIClient,
+};
+
+/// Status object for shipcatmanifests crd
+///
+/// All fields optional, but we try to ensure all fields exist.
+#[derive(Deserialize, Serialize, Clone, Debug, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct ManifestStatus {
+    /// Detailed individual conditions, emitted as they happen during apply
+    pub conditions: Option<Conditions>,
```

Comment:
Can this not be defaulted?

```suggestion
	#[serde(default)]
    pub conditions: Conditions,
```

Because

```rs
None == Some(Conditions { apply: None, rollout: None }) == Some(Conditions::default())
```

Therefore there's no need to unwrap the option elsewhere

Reply:
Just because it has a default doesn't mean it's sensible to treat a missing value as such. A default condition would imply a concrete timestamp here (which I don't want).

Reply:
A default `Condition` has a timestamp, but a default `Conditions` has `None` for each `Condition`

Reply:
Ah. true!